src/utils/DependencyUtils.ts

```diff
@@ -11,3 +34,100 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+/**
+ * Dependency resolution varies based on whether packages are being
+ * installed or updated, and whether the package is a mod or a modpack.
+ *
+ * 1. When installing a mod
```

Comment:
This referring to a singular mod and not multiple is a bit misleading. While the current usage in the app is only either a singular mod, or updating multiple, new features like retrying make installing multiple specific versions, with latest dependencies. (the logic itself seems to work correctly, but the docstring doesn't explain what happens with multiple)

Reply:
My intention was to write the whole docstring from the perspective of "current usage in the app". I guess I could try to make that more explicit and add a note about other usage. Also I'll reconsider the wording to make it more exact than "shoulds".